Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/mysql/MySQLNode.java

```diff
@@ -0,0 +1,124 @@
+import com.microsoft.azuretools.telemetry.AppInsightsConstants;
+import com.microsoft.azuretools.telemetry.TelemetryProperties;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.microsoft.tooling.msservices.serviceexplorer.*;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.validator.routines.InetAddressValidator;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.microsoft.azuretools.telemetry.TelemetryConstants.ENABLE_MY_PUBLIC_IP;
+import static com.microsoft.azuretools.telemetry.TelemetryConstants.MYSQL;
+import static com.microsoft.tooling.msservices.serviceexplorer.azure.mysql.MySQLModule.ICON_FILE;
+
+
+public class MySQLNode extends Node implements TelemetryProperties {
+    private static final InetAddressValidator ipValidator = InetAddressValidator.getInstance();
+    private static final String ACTION_ADD_MY_IP = "Add My IP";
+    private final String subscriptionId;
+    private final Server server;
+
+    public MySQLNode(AzureRefreshableNode parent, String subscriptionId, Server mysqlServer) {
+        super(mysqlServer.id(), mysqlServer.name(), parent, ICON_FILE, true);
+        this.subscriptionId = subscriptionId;
+        this.server = mysqlServer;
+        loadActions();
+    }
+
+    public Map<String, String> toProperties() {
+        final Map<String, String> properties = new HashMap<>();
+        properties.put(AppInsightsConstants.SubscriptionId, this.subscriptionId);
+        // todo: track region name
+        return properties;
+    }
+
+    @Override
+    protected void loadActions() {
+        addAction(ACTION_ADD_MY_IP, new WrappedTelemetryNodeActionListener(MYSQL, ENABLE_MY_PUBLIC_IP,
+                                                                           createBackgroundActionListener("Enabling",
+                                                                               () -> enableAccessFromMyMachine())));
+    }
+
+    private static NodeActionListener createBackgroundActionListener(final String actionName, final Runnable runnable) {
+        return new NodeActionListener() {
+            @Override
+            protected void actionPerformed(NodeActionEvent e) {
+                DefaultLoader.getIdeHelper().runInBackground(null, actionName, false,
+                                                             true, String.format("%s...", actionName), runnable);
+            }
+        };
+    }
+
+    private void enableAccessFromMyMachine() {
+        String myIp = getIpAddress();
+        if (StringUtils.isBlank(myIp)) {
+            DefaultLoader.getUIHelper().showErrorNotification("Cannot get public ip",
+                    "Cannot get public ip address, please make sure you are online and can access http://bot.whatismyipaddress.com/");
```

Comment:
Is this a common method to get local machine public ip? Shall we change to ping `8.8.8.8`?

By the way, could this also works in VPN?

Reply:
https://docs.microsoft.com/en-us/azure/developer/java/spring-framework/configure-spring-data-jdbc-with-azure-mysql

Reply:
will replace to http://whatismyip.akamai.com/